### PR TITLE
Update manual picker

### PR DESCRIPTION
**Pull request by SteffenHeu**
as discussed on slack

- double click on area type opens the dialog
- update everything to the new feature model

### CODE REVIEW

File: src/main/java/io/github/mzmine/datamodel/features/types/numbers/AreaType.java [hunk 1]
```diff
@@ -42,4 +50,20 @@ public String getUniqueID() {
...
+  public @Nullable Runnable getDoubleClickAction(@NotNull ModularFeatureListRow row,
+      @NotNull List<RawDataFile> file, @Nullable DataType<?> superType, @Nullable Object value) {
+
+    if(file.size() == 1) {
+      final ModularFeature selectedFeature = row.getFeature(file.get(0));
+
+      if(selectedFeature != null && selectedFeature.getFeatureStatus() != FeatureStatus.UNKNOWN) {
+        return () -> MZmineCore.runLater(() -> XICManualPickerModule.runManualDetection(selectedFeature.getRawDataFile(),
+            row, row.getFeatureList()));
+      }
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/datamodel/features/types/numbers/AreaType.java:64]: could we eneable this somehow for data files where the feature was undetected?

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_manual/ManualPickerTask.java [hunk 2]
```diff
@@ -42,15 +49,15 @@
 import java.util.Date;
 import java.util.List;
 import java.util.logging.Logger;
+import org.bouncycastle.math.raw.Mod;
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_manual/ManualPickerTask.java:52]: what is this?

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_manual/ManualPickerTask.java [hunk 3]
```diff
@@ -93,51 +102,16 @@ public void run() {
...
-        newFeature.finalizeFeature();
-        newFeature.setFeatureList(featureList);
-        if (newFeature.getArea() > 0) {
-          featureListRow.addFeature(dataFile,
-              FeatureConvertors.ManualFeatureToModularFeature(featureList, newFeature));
-        }
-      } else {
-        featureListRow.removeFeature(dataFile);
-      }
+    final ScanSelection selection = new ScanSelection(rtRange, 1);
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_manual/ManualPickerTask.java:105]: The list of scans is previously known to the feature list - so maybe just use the complete list of scans? this way there are no issues with polarity switching and other weird scan selections

File: src/main/java/io/github/mzmine/util/FeatureUtils.java [hunk 4]
```diff
@@ -241,34 +246,11 @@ public static Range<Double> findMZRange(Feature[] features) {
    */
   public static double integrateOverMzRtRange(RawDataFile dataFile, Range<Float> rtRange,
       Range<Double> mzRange) {
-    ManualFeature newFeature = new ManualFeature(dataFile);
-    boolean dataPointFound = false;
+    ScanSelection selection = new ScanSelection(rtRange, 1);
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/util/FeatureUtils.java:249]: any way to pass in the actual scan selection?

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_manual/ManualPickerTask.java (hunk 2 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/dataprocessing/featdet_manual/ManualPickerTask.java:52]: the thing i import when i press "tab" too fast when typing "ModularFeature" :D

File: src/main/java/io/github/mzmine/datamodel/features/types/numbers/AreaType.java (hunk 1 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/datamodel/features/types/numbers/AreaType.java:64]: probably, but no time to test that at the moment

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_manual/ManualPickerTask.java (hunk 3 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/dataprocessing/featdet_manual/ManualPickerTask.java:105]: good point, fixed

File: src/main/java/io/github/mzmine/util/FeatureUtils.java (hunk 4 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/util/FeatureUtils.java:249]: done

**Review by robinschmid — APPROVED** [review]
Looks great

**Pull request merged**